crates/wasm-mutate/src/mutators/peephole/eggsy/encoder.rs

```diff
@@ -410,23 +409,67 @@ impl Encoder {
         info: &ModuleInfo,
         egraph: &EG,
         entry: &StackEntry,
-        _: usize,
+        operators: &Vec<OperatorAndByteOffset>,
         newfunc: &mut Function,
     ) -> crate::Result<()> {
         // Write the deps in the dfg
         // Process operands
-        if entry.is_undef {
+        if let StackType::Undef = entry.operator {
             // Do nothing
             // log
         } else {
             for idx in &entry.operands {
                 let entry = &egraph.analysis.get_stack_entry(*idx);
-                Encoder::writestackentry(info, egraph, entry, *idx, newfunc)?;
+                Encoder::writestackentry(info, egraph, entry, operators, newfunc)?;
             }
             // Write the operator
-            let bytes = &info.get_code_section().data
-                [entry.byte_stream_range.start..entry.byte_stream_range.end];
-            newfunc.raw(bytes.iter().copied());
+
+            match entry.operator {
+                StackType::I32(val) => {
+                    newfunc.instruction(Instruction::I32Const(val));
+                }
+                StackType::I64(val) => {
+                    newfunc.instruction(Instruction::I64Const(val));
+                }
+                StackType::LocalGet(idx) => {
+                    newfunc.instruction(Instruction::LocalGet(idx));
+                }
+                StackType::LocalSet(idx) => {
+                    newfunc.instruction(Instruction::LocalSet(idx));
+                }
+                StackType::Load(static_offset, align, idx) => {
+                    // Here it depends on the type
+                    // TODO add the other operands here as well
+                    match entry.tpes[..] {
+                        [PrimitiveTypeInfo::I32] => {
+                            newfunc.instruction(Instruction::I32Load(MemArg {
+                                offset: static_offset,
+                                align: align as u32,
+                                memory_index: idx,
+                            }));
+                        }
+                        [PrimitiveTypeInfo::I64] => {
+                            newfunc.instruction(Instruction::I64Load(MemArg {
+                                offset: static_offset,
+                                align: align as u32,
+                                memory_index: idx,
+                            }));
+                        }
+                        _ => panic!("Replace with err"),
+                    }
+                }
```

Comment:
I don't think it is generally safe to re-encode the load since that might move it across a function call or a store that can change the value at the loaded address, leading to different behavior.